R/duckreg.R

```diff
@@ -2,220 +2,731 @@
-#' the default.
-#' @param query_only Logical indicating whether only the underlying compression
-#'   SQL query should be returned (i.e., no computation will be performed).
-#'   Default is `FALSE`.
-#' @param data_only Logical indicating whether only the compressed dataset
-#'   should be returned (i.e., no regression is run). Default is `FALSE`.
-#' 
-#' @return A list of class "duckreg" containing various slots, including a table
-#' of coefficients (which the associated print method will display).
-#' @references
-#' Wong, J., Forsell, E., Lewis, R., Mao, T., & Wardrop, M. (2021).
-#' \cite{You Only Compress Once: Optimal Data Compression for Estimating Linear Models.} 
-#' arXiv preprint arXiv:2102.11297.
-#' Available: https://doi.org/10.48550/arXiv.2102.11297
-#' 
+#' Internal strategies:
+#' 1. group: GROUP BY regressors (+ fixed effects) -> frequency-weighted rows -> WLS.
+#' 2. moments: Global sufficient statistics (X'X, X'y) (no fixed effects).
+#' 3. moments_fe: One or two-way fixed-effect within (demean/double-demean) + cross-moments.
+#'
+#' Auto heuristic:
+#' - If no FEs and (any regressor appears continuous OR estimated compression ratio > compression_threshold) => moments.
+#' - If 1-2 FEs and estimated compression ratio high (> max(0.6, compression_threshold)) => moments_fe.
+#' - Else group.
+#'
+#' Notes:
+#' - Robust HC1: group path computes HC1; moments/moments_fe apply a df correction to classical variance.
+#'
+#' @param fml Formula. Use pipe for FEs: y ~ x1 + x2 | fe1 + fe2
+#' @param conn DuckDB connection (optional).
+#' @param table Character table name already in conn, or tbl_lazy object.
+#' @param data Data frame to register (ignored if table given).
+#' @param path DuckDB table function or path string.
+#' @param vcov "hc1" or "ols".
+#' @param query_only Return only SQL (group strategy only).
+#' @param data_only Return compressed rows (group strategy only).
+#' @param strategy "auto","group","moments","moments_fe".
+#' @param compression_threshold Threshold for switching away from group in auto.
+#' @param verbose Logical diagnostics.
+#' @param ridge_rel Relative ridge (lambda = ridge_rel * mean(diag(X'X))) if Cholesky fails.
+#'
+#' @return List (class duckreg) with: coeftable, vcov, strategy, compression info, etc.
+#' @references Wong et al. (2021) arXiv:2102.11297
 #' @importFrom DBI dbConnect dbDisconnect dbGetQuery
 #' @importFrom duckdb duckdb duckdb_register
 #' @importFrom Formula Formula
 #' @importFrom Matrix chol2inv crossprod Diagonal sparse.model.matrix 
-#' @importFrom stats reformulate
-#' @importFrom glue glue glue_sql
-#' 
+#' @importFrom stats reformulate pt
+#' @importFrom glue glue
+#' @export
+duckreg <- function(
+  fml,
+  conn = NULL,
+  table = NULL,
+  data = NULL,
+  path = NULL,
+  vcov = "hc1",
+  query_only = FALSE,
+  data_only = FALSE,
+  strategy = c("auto","group","moments","moments_fe"),
+  compression_threshold = 0.001,
```

Comment:
nit: let's just call it `threshold` (or, to use a more idiomatic R verb: `tol`?)